src/dll/Hooks/ExecuteProcess.cpp

```diff
@@ -92,16 +92,17 @@ bool ExecuteScc(SccApi& scc)
     auto r6Dir = App::Get()->GetPaths()->GetR6Dir();
     auto scriptSystem = App::Get()->GetScriptCompilationSystem();
 
-    auto settings = scc.settings_new(r6Dir.string().c_str());
+    auto settings = scc.settings_new(reinterpret_cast<const char*>(r6Dir.u8string().c_str()));
 
     if (scriptSystem->HasScriptsBlob())
     {
-        scc.settings_set_custom_cache_file(settings, scriptSystem->GetScriptsBlob().string().c_str());
+        auto blobPath = reinterpret_cast<const char*>(scriptSystem->GetScriptsBlob().u8string().c_str());
+        scc.settings_set_custom_cache_file(settings, blobPath);
```

Comment:
this is wrong, you get an internal address to temporary u8string here which is deallocated right after

the pointer you get here points to freed memory.

you have to store the string in some variable and do the cast then with that...

or do it all in one go like you did few lines above.

Reply:
temporaries like one produced by the call u8string live "just enough" in C++ to finish a statement they are used in. so after this statement ended, the string returned via u8string is freed, therefore, pointer you got from that string earlier is no longer valid.

just for some better explanation why this is bad.

Reply:
btw, this is only line you split for some reason (probably because of the length), other changes are fine as mentioned because they are all in one statement.